src/Algebra/Structures.agda

```diff
@@ -415,15 +415,22 @@ record IsSemiringWithoutOne (+ * : Op₂ A) (0# : A) : Set (a ⊔ ℓ) where
     zero                  : Zero 0# *
 
   open IsCommutativeMonoid +-isCommutativeMonoid public
-    using (setoid)
+    using (setoid; isEquivalence)
     renaming
-    ( comm                   to +-comm
+    ( ∙-cong                 to +-cong
+    ; ∙-congˡ                to +-congˡ
+    ; ∙-congʳ                to +-congʳ
+    ; assoc                  to +-assoc
+    ; identity               to +-identity
+    ; identityˡ              to +-identityˡ
+    ; identityʳ              to +-identityʳ
+    ; comm                   to +-comm
     ; isMonoid               to +-isMonoid
     ; isCommutativeMagma     to +-isCommutativeMagma
     ; isCommutativeSemigroup to +-isCommutativeSemigroup
     )
 
-  open Setoid setoid public
+  open IsEquivalence isEquivalence public
```

Comment:
Doesn't this removal of `open Setoid setoid public` change the export, i.e. `Carrier` is no longer visible?

Reply:
Yes, this is intentional - no other structure exposes `Carrier`, and doing so was causing issues when trying to use the corresponding bundle.

Reply:
See longer comment below

Reply:
Indeed, no (other) `Structure` exports `setoid`, nor I think, should.

Reply:
Re: the export of `setoid` from `IsMagma`: this is used by `IsGroup` in this module, so can't be removed at this stage.